src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx

```diff
@@ -0,0 +1,52 @@
+import codeBlockPrimitives from '@hashicorp/react-code-block/mdx'
+import { CodeTabsProps } from '@hashicorp/react-code-block/partials/code-tabs'
+import classNames from 'classnames'
+import s from './mdx-code-blocks.module.css'
+
+/**
+ * In all MDX contexts in Dev Dot, we want our
+ * MDX code block primitives to have a dark theme.
+ */
+const {
+	CodeBlockConfig,
+	CodeTabs,
+	pre: ReactPre,
+} = codeBlockPrimitives({
+	theme: 'dark',
+})
+
+/**
+ * Re-exports react-code-block <CodeBlockConfig />, for naming consistency.
+ */
+export const MdxCodeBlockConfig = CodeBlockConfig
+
+/**
+ * Adds spacing specific to Dev Dot to react-code-block <CodeTabs/>.
+ */
+export function MdxCodeTabs({ className, ...restProps }: CodeTabsProps) {
+	return (
+		<CodeTabs
+			{...restProps}
+			className={classNames(className, s.codeTabsMargin)}
+		/>
+	)
+}
+
+/**
+ * Adds spacing specific to Dev Dot to the react-code-block <pre> MDX component.
```

Comment:
I like this, better than adjusting the core component and aligns with our other patterns of creating _mdx_ specific instances for authorable components.